react/src/components/ManageAppsModal.tsx

```diff
@@ -1,101 +1,256 @@
+    useMutation<ManageAppsModalMutation>(graphql`
+      mutation ManageAppsModalMutation(
+        $target: String!
+        $architecture: String
+        $props: ModifyImageInput!
+      ) {
+        modify_image(
+          target: $target
+          architecture: $architecture
+          props: $props
+        ) {
+          ok
+          msg
+        }
+      }
+    `);
+
   let parsedValue: {
-    open: boolean;
+    image: any;
     servicePorts: any;
   };
   try {
     parsedValue = JSON.parse(value || '');
   } catch (error) {
     parsedValue = {
-      open: false,
+      image: {},
       servicePorts: [],
     };
   }
-  const { open, servicePorts } = parsedValue;
-  const onFinish = (values: any) => {
-    console.log('Saved settings. ', values);
+  const { image, servicePorts } = parsedValue;
+
+  const handleOnclick = async () => {
+    try {
+      await form.validateFields();
+
+      const values = form
+        .getFieldValue('apps')
+        .map((item: { app: string; protocol: string; port: number }) => {
+          return `${item.app}:${item.protocol}:${item.port}`;
+        })
+        .join(',');
+
+      const INPUT: KVPairInput[] = [];
+      Object.entries(image.labels).forEach(([key, value]) => {
+        if (key.includes('service-ports')) {
+          INPUT.push({ key: key, value: values });
+        } else {
+          INPUT.push({ key: key, value: value?.toString() || '' });
+        }
+      });
+
+      const commitRequest = () =>
+        commitModifyImageInput({
+          variables: {
+            target: `${image.registry}/${image.name}:${image.tag}`,
+            architecture: image.architecture,
+            props: {
+              labels: INPUT,
+              resource_limits: null,
+            },
+          },
+          onCompleted: (res, err) => {
+            console.log(res, err);
+            message.success(t('environment.DescImagePortsModified'));
+            dispatchEvent('ok', null);
+            return;
+          },
+          onError: (err) => {
+            console.log(err);
+            message.error(t('dialog.ErrorOccurred'));
+          },
+        });
+
+      if (image.installed) {
+        app.modal.confirm({
+          title: 'Image reinstallation required',
+          content: (
+            <>
+              <Trans
+                i18nKey={
+                  'envrionment.ModifyImageResourceLimitReinstallRequired'
+                }
+              />
+            </>
+          ),
+          onOk: commitRequest,
+          getContainer: () => document.body,
+          closable: true,
+        });
+      } else {
+        commitRequest();
+      }
+    } catch (info: any) {
+      setValidateDetail(info.errorFields[0].errors[0]);
+      return;
+    }
   };
 
   return (
     <BAIModal
+      destroyOnClose
       open={open}
-      onCancel={() => {
-        dispatchEvent('cancel', null);
-      }}
-      centered
+      onOk={handleOnclick}
```

Comment:
Please rename like this :)
```suggestion
      onOk={handleOnClick}
```